simtools/corsika/corsika_config.py

```diff
@@ -697,3 +433,49 @@ def get_corsika_input_list(self):
             corsika_input_list += f"\t # {telescope_name}\n"
 
         return corsika_input_list
+
+    @property
+    def run_number(self):
+        """Set run number."""
+        return self._run_number
+
+    @run_number.setter
+    def run_number(self, run_number):
+        """
+        Set run number and validate it.
+
+        Parameters
+        ----------
+        run_number: int
+            Run number.
+        """
+        self._run_number = self.validate_run_number(run_number)
+
+    def validate_run_number(self, run_number):
+        """
+        Validate run number and return it.
+
+        Return run number from configuration if None.
+
+        Parameters
+        ----------
+        run_number: int
+            Run number.
+
+        Returns
+        -------
+        int
+            Run number.
+
+        Raises
+        ------
+        ValueError
+            If run_number is not a valid value (e.g., < 1).
+        """
+        if run_number is None:
+            return self.run_number
+        if not float(run_number).is_integer() or run_number < 1 or run_number > 999999:
```

Comment:
Can we have several runs simultaneausly if we use divergent pointing?

Reply:
I am not sure I understand your comment. Can you explain a little bit what you mean? Why is this connected to divergent pointing?

Reply:
Maybe having >200 runs per night (if we define several runs parallel), can lead to exceeding this limit over the lifetime of CTAO. Not sure this is realistic though.

Reply:
This is a corsika limit! Nothing we can do about, corsika does not allow larger run numbers (Note that there is no meaning for run numbers, we can simulate all runs with the same run number and things would be ok).

Reply:
Ah ok I see. Then all good here.